### PR TITLE
Enhancement/multiple fdb instances

### PR DESCRIPTION
**Pull request by geier1993**
PR for feature request #17

Still needs testing  and discussion

### CODE REVIEW

**Comment by FussyDuck** [comment]
[CLA assistant check](https://bol-claassistant.ecmwf.int:443/ecmwf/pyfdb?pullRequest=18) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://bol-claassistant.ecmwf.int:443/ecmwf/pyfdb?pullRequest=18) before we can accept your contribution.<br/>**2** out of **4** committers have signed the CLA.<br/><br/>:white_check_mark: kinow<br/>:white_check_mark: tlmquintino<br/>:x: oiffrig<br/>:x: geier1993<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://bol-claassistant.ecmwf.int:443/check/ecmwf/pyfdb?pullRequest=18) it.</sub>

File: pyfdb/pyfdb.py [hunk 1]
```diff
@@ -260,10 +260,17 @@ class FDB:
...
 
-    def __init__(self):
+    def __init__(self, config_path=None):
         fdb = ffi.new('fdb_handle_t**')
-        lib.fdb_new_handle(fdb)
-
+        if config_path is None:
+            lib.fdb_new_handle(fdb)
+        else:
+            if type(config_path) is str:
```
Comment (simondsmart) [inline pyfdb/pyfdb.py:268]: My instinct is that this is not the right way to do this. I think we probably want to support the Config object (or at the least a LocalConfiguration) natively in the C API, and accept that as an argument.

We also need to handle the userConfiguration correctly.

@geier1993 - can you recall what (if any) support exists for the LocalConfig object via C APIs?

Comment (geier1993) [reply pyfdb/pyfdb.py:268]: There is no C API for eckit AFAIK.

**Comment by tbkr** [comment]
This should now be fixed with https://github.com/ecmwf/pyfdb/pull/25. Feel free to reopen in case the feature isn't fully covering your requirements.

**Pull request closed without merging**